services/galley/src/Galley/API/MLS/Removal.hs

```diff
@@ -110,9 +110,10 @@ removeClient ::
   ClientId ->
   Sem r ()
 removeClient lc qusr cid = do
```

Comment:
Can you see if you can change the signature of the function such that you don't need to do `for_` in the function? You know you do this only for MLS clients, so it feels a bit awkward to have to check if this is an MLS conversation.

I think you could pass in `ConversationMLSData` and as little as needed from `Local Data.Conversation` instead of passing in the whole conversation.

Reply:
Downstream of `removeClient` is `propagateMessage` and `runMessagePush` which require the full conversation object. Maybe we could refactor them as well, but I woudn't want to mix it into this PR

Reply:
Fair point about inlining the `for_` part into `removeUser` itself. Did this is acf5d67bb0fc9e61753e927bb8f7751377a33c26